image-loader/test/scala/model/ImageUploadTest.scala

```diff
@@ -0,0 +1,177 @@
+
+    def storeOrProjectOriginalFile: StorableOriginalImage => Future[S3Object] = {
+      a => {
+        Future.successful(
+          mockS3Object
+            .copy(size = a.file.length())
+            .copy(metadata = mockS3Object.metadata
+              .copy(objectMetadata = mockS3Object.metadata.objectMetadata
+                .copy(contentType = Some(a.mimeType)))))
+      }
+    }
+
+    def storeOrProjectThumbFile: StorableThumbImage => Future[S3Object] = {
+      a => {
+        Future.successful(
+          mockS3Object
+            .copy(size = a.file.length())
+            .copy(metadata = mockS3Object.metadata
+              .copy(objectMetadata = mockS3Object.metadata.objectMetadata
+                .copy(contentType = Some(a.mimeType))))
+        )
+      }
+    }
+
+    def storeOrProjectOptimisedPNG: StorableOptimisedImage => Future[S3Object] = {
+      a => {
+        if (a.mimeType == Jpeg) fail("We should not create optimised jpg")
+        Future.successful(
+          mockS3Object
+            .copy(size = a.file.length())
+            .copy(metadata = mockS3Object.metadata
+              .copy(objectMetadata = mockS3Object.metadata.objectMetadata
+                .copy(contentType = Some(a.mimeType))))
+        )
+      }
+    }
+
+    val mockDependencies = ImageUploadOpsDependencies(
+      mockConfig,
+      imageOps,
+      storeOrProjectOriginalFile,
+      storeOrProjectThumbFile,
+      storeOrProjectOptimisedPNG
+    )
+
+    val tempFile = ResourceHelpers.fileAt(fileName)
+    val ul = UploadInfo(None)
+
+    val uploadRequest = UploadRequest(
+      uuid,
+      randomId,
+      tempFile,
+      MimeTypeDetection.guessMimeType(tempFile).right.toOption,
+      DateTime.now(),
+      "uploadedBy",
+      Map(),
+      ul
+    )
+
+    val futureImage = Uploader.uploadAndStoreImage(
+      mockDependencies.storeOrProjectOriginalFile,
+      mockDependencies.storeOrProjectThumbFile,
+      mockDependencies.storeOrProjectOptimisedImage,
+      OptimiseWithPngQuant,
+      uploadRequest,
+      mockDependencies,
+      FileMetadata()
+    )
+
+    // Assertions; Failure will auto-fail
+    futureImage.map(i => {
+      // Assertions on original request
+      assert(i.id == randomId, "Correct id comes back")
+      assert(i.source.mimeType.contains(expectedOriginalMimeType), "Should have the correct mime type")
+
+      // Assertions on generated thumbnail image
+      assert(i.thumbnail.isDefined, "Should always create a thumbnail")
+      assert(i.thumbnail.get.mimeType.get == Jpeg, "Should have correct thumb mime type")
+
+      // Assertions on optional generated optimised png image
+      assert(i.optimisedPng.isDefined == expectOptimisedFile, "Should have optimised file")
+      assert(!expectOptimisedFile || i.optimisedPng.flatMap(p => p.mimeType).contains(Png), "Should have correct optimised mime type")
+    })
+  }
+
+  ignore("rubbish") {
```

Comment:
@aug24 it'd be worth stating what we're testing for here, I'll have a try.

Reply:
Could we make the test name describe what it's testing? `rubbish` or `tiff_8bpc_layered_withTransparency` etc. is too opaque imo.